### PR TITLE
Automate updates to `CHANGELOG-old.md`

### LINKED ISSUE #2726: Automate updates to `CHANGELOG-old.md`
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Copying of the old CHANGELOG to the new one is not automated.
**Describe the solution you'd like**

Modify `dev/release/update_change_log.sh` so that this is done when the new CHANGELOG is generated.
**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by iajoiner**
# Which issue does this PR close?

Closes #2726.

# Rationale for this change
 

# What changes are included in this PR?
Automated update to `CHANGELOG-old.md`

# Are there any user-facing changes?

### CODE REVIEW

**Comment by iajoiner** [comment]
@alamb  Here we go!

**Comment by alamb** [comment]
Thanks @iajoiner

File: dev/release/update_change_log.sh [hunk 1]
```diff
@@ -29,16 +29,45 @@
...
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Historical Changelog
+' | cat - "${OUTPUT_PATH}" "${OLD_OUTPUT_PATH}" > "${OLD_OUTPUT_PATH}".tmp
+mv "${OLD_OUTPUT_PATH}".tmp "${OLD_OUTPUT_PATH}"
```
Comment (alamb) [inline dev/release/update_change_log.sh:70]: Shall we maybe also remove `"${OLD_OUTPUT_PATH}".tmp`?

**Review by alamb — APPROVED** [review]

**Comment by iajoiner** [comment]
@alamb I don’t think the random, likely transient error has anything to do with the PR.

**Comment by iajoiner** [comment]
@alamb Yeah it is a Rust issue and I have filed it over there: https://github.com/rust-lang/rust/issues/101869

**Pull request merged**